src/libraries/Common/src/Interop/Windows/Kernel32/Interop.GetProcessName.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class Kernel32
+    {
+        [DllImport(Libraries.Kernel32, CharSet = CharSet.Unicode, EntryPoint = "QueryFullProcessImageNameW", ExactSpelling = true)]
+        private static extern bool QueryFullProcessImageName(
+            IntPtr hProcess,
+            uint dwFlags,
+            ref char lpBuffer,
+            ref uint lpdwSize);
+
+        private const int MAX_PROCESSNAME_LENGTH = 1024;
+
+        internal static string? GetProcessName(uint processId)
+        {
+            using (SafeProcessHandle h = Interop.Kernel32.OpenProcess(Interop.Advapi32.ProcessOptions.PROCESS_QUERY_LIMITED_INFORMATION, false, (int)processId))
+            {
+                Span<char> buffer = stackalloc char[MAX_PROCESSNAME_LENGTH + 1];
+                uint length = (uint)buffer.Length;
+
+                bool queried = QueryFullProcessImageName(
+                    h.DangerousGetHandle(),
```

Comment:
It would be safer to use `SafeProcessHandle` and let the marshaller do it's job rather than using `DangerousGetHandle()`:

https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle.dangerousgethandle?view=net-5.0#remarks

```suggestion
            SafeHandle hProcess,
            uint dwFlags,
            ref char lpBuffer,
            ref uint lpdwSize);

        private const int MAX_PROCESSNAME_LENGTH = 1024;

        internal static string? GetProcessName(uint processId)
        {
            using (SafeProcessHandle h = Interop.Kernel32.OpenProcess(Interop.Advapi32.ProcessOptions.PROCESS_QUERY_LIMITED_INFORMATION, false, (int)processId))
            {
                Span<char> buffer = stackalloc char[MAX_PROCESSNAME_LENGTH + 1];
                uint length = (uint)buffer.Length;

                bool queried = QueryFullProcessImageName(
                    h,
```